src/main.rs

```diff
@@ -868,35 +864,28 @@ fn real_main() -> Result<(), VetError> {
         }),
     };
 
-    let workspace_metacfg = || -> Option<MetaConfigInstance> {
-        // FIXME: what is `store.path` relative to here?
-        MetaConfigInstance::deserialize(metadata.workspace_metadata.get(WORKSPACE_VET_CONFIG)?)
-            .map_err(|e| {
-                error!(
-                    "Workspace had [{WORKSPACE_VET_CONFIG}] but it was malformed: {}",
-                    e
-                );
-                panic_any(ExitPanic(-1));
-            })
-            .ok()
-    }();
-
-    let package_metacfg = || -> Option<MetaConfigInstance> {
-        // FIXME: what is `store.path` relative to here?
-        MetaConfigInstance::deserialize(metadata.root_package()?.metadata.get(PACKAGE_VET_CONFIG)?)
-            .map_err(|e| {
-                error!(
-                    "Root package had [{PACKAGE_VET_CONFIG}] but it was malformed: {}",
-                    e
-                );
-                panic_any(ExitPanic(-1));
-            })
-            .ok()
-    }();
+    // FIXME: what is `store.path` relative to here?
+    let workspace_metacfg = metadata
+        .workspace_metadata
+        .get(WORKSPACE_VET_CONFIG)
+        .map(|cfg| {
+            MetaConfigInstance::deserialize(cfg)
+                .wrap_err("Workspace had [{WORKSPACE_VET_CONFIG}] but it was malformed")
+        })
+        .transpose()?;
+
+    // FIXME: what is `store.path` relative to here?
+    let package_metacfg = metadata
+        .root_package()
+        .and_then(|r| r.metadata.get(PACKAGE_VET_CONFIG))
+        .map(|cfg| {
+            MetaConfigInstance::deserialize(cfg)
+                .wrap_err("Root package had [{PACKAGE_VET_CONFIG}] but it was malformed")
+        })
+        .transpose()?;
```

Comment:
I changed these to use closures still initially and didn't like it personally so I pushed a second commit to change these to use method chaining, if you prefer the old it should be easy to back this out and go back to a version more consistent with what was in place before, and I can also change the return type on the closures to be just `Option<_>` instead of the full type which I wrote in in the previous commit

Reply:
This is way nicer, the closures were me experiencing defeat at adapter-foo (TIL about transpose!).